source/common/stats/metric_impl.h

```diff
@@ -30,7 +32,6 @@ class MetricImpl : public virtual Metric {
   };
 
 private:
-  const std::string name_;
   const std::string tag_extracted_name_;
```

Comment:
It looks like we could get rid of this too; we'd have to change consumers of tagExtractedName() and tags() to compute them on demand from TagProducerImpl::produceTags(). That could be a separate PR.

I don't know what the CPU cost of that would be; if that happens during operation, or would be only while setting up a stats sinks.  @mrice32 , WDYT?

If not, the tag_extracted_name_ string would be ripe for symbolizing later.

Reply:
I'm OK with spinning this off into a separate PR, but yeah, if we do this for one variable we might well do it for several.